artificial-scrum-master/apps/artificial.scrum.master/Artificial.Scrum.Master.ScrumProjectIntegration/Infrastructure/ApiTokens/TokenValidator.cs

```diff
@@ -0,0 +1,24 @@
+using Artificial.Scrum.Master.ScrumProjectIntegration.Utilities;
+
+namespace Artificial.Scrum.Master.ScrumProjectIntegration.Infrastructure.ApiTokens
+{
+    public class TokenValidator : ITokenValidator
+    {
+        private const int TokenRefreshRequestTimeBuffer = 2;
+        private const string ExpirationTimeClaimTypeName = "exp";
+
+        private readonly IJwtDecoder _jwtDecoder;
+
+        public TokenValidator(IJwtDecoder jwtDecoder)
+        {
+            _jwtDecoder = jwtDecoder;
+        }
+
+        public bool ValidateAccessTokenExpirationTime(string accessToken)
+        {
+            var expirationDate = _jwtDecoder.GetExpirationDate(accessToken, ExpirationTimeClaimTypeName);
+            var expired = DateTime.UtcNow.AddMinutes(-TokenRefreshRequestTimeBuffer);
```

Comment:
sorry, że dopiero teraz ale mógłbyś skorzystać tutaj z wstrzykniętej klasy https://learn.microsoft.com/pl-pl/dotnet/api/system.timeprovider?view=net-8.0 i w infrastructure w DI dodać services.AddTransient(TimeProvider.Default)? Dzięki temu będzie można to zmockować w testach